### DISCUSSION TITLE
Struggling with blocking calls

### DISCUSSION THREAD

**Discussion by gannj001** [supersaiyanmode/PyWebOSTV#58] [Q&A]
I'm trying to start an app, and then wait until it's opened. Any insight on how to do this? Currently I have this:
```
    apps = app.list_apps()
    yt = [x for x in apps if "youtube" in x["title"].lower()][0]
    launch_info = app.launch(yt)
    system.notify("I launched YouTube!")
    print(launch_info)
    system.notify("Closing Youtube")
    app.close(launch_info)
```

**Comment by supersaiyanmode** [comment]
Unfortunately, I don't think WebOS APIs provides us a way to wait until the app has fully been loaded. You can maybe use `app.get_current()`  to query the foreground app (or it's subscribe equivalent to notify you), but I am not entirely sure if this is able to distinguish between app loading state and app ready state.
